pkg/services/app/add.go

```diff
@@ -240,58 +258,57 @@ func (a *App) getGitRemoteUrl(params AddParams) (string, error) {
 	return sanitizeRepoUrl(urls[0]), nil
 }
 
-func (a *App) addAppWithNoConfigRepo(params AddParams, secretRef string, appHash string) error {
+func (a *App) addAppWithNoConfigRepo(info *AppResourceInfo, dryRun bool, secretRef string, appHash string) error {
 	// Returns the source, app spec and kustomization
-	source, appGoat, appSpec, err := a.generateAppManifests(params, secretRef, appHash)
+	source, appGoat, appSpec, err := a.generateAppManifests(info, secretRef, appHash)
 	if err != nil {
 		return fmt.Errorf("could not generate application GitOps Automation manifests: %w", err)
 	}
 
 	a.logger.Actionf("Applying manifests to the cluster")
-	return a.applyToCluster(params, source, appGoat, appSpec)
+	return a.applyToCluster(info, dryRun, source, appGoat, appSpec)
 }
 
-func (a *App) addAppWithConfigInAppRepo(params AddParams, gitProvider gitproviders.GitProvider, clusterName string, secretRef string, appHash string) error {
+func (a *App) addAppWithConfigInAppRepo(info *AppResourceInfo, params AddParams, gitProvider gitproviders.GitProvider, secretRef string, appHash string) error {
 	// Returns the source, app spec and kustomization
-	source, appGoat, appSpec, err := a.generateAppManifests(params, secretRef, appHash)
+	source, appGoat, appSpec, err := a.generateAppManifests(info, secretRef, appHash)
 	if err != nil {
 		return fmt.Errorf("could not generate application GitOps Automation manifests: %w", err)
 	}
 
-	// Kustomization pointing to the repo in .wego directory
-	appWegoGoat, err := a.generateAppWegoManifests(params, clusterName)
+	appWegoGoat, err := a.generateAppWegoManifests(info)
 	if err != nil {
 		return fmt.Errorf("could not create GitOps automation for .wego directory: %w", err)
 	}
 
 	// a local directory has not been passed, so we clone the repo passed in the --url
 	if params.Dir == "" {
-		a.logger.Actionf("Cloning %s", params.Url)
-		if err := a.cloneRepo(params.Url, params.Branch, params.DryRun); err != nil {
+		a.logger.Actionf("Cloning %s", info.Spec.URL)
+		if err := a.cloneRepo(info.Spec.URL, info.Spec.Branch, params.DryRun); err != nil {
```

Comment:
would it be a good idea to remove the cloned repo when the user uses `--url` flag ? The cloning is happening in a temporary folder but I wonder if we should delete that temporary folder right away or if it would be ok to let the system delete it later? 🤔

Reply:
Wow. Funny we never noticed that before. :+1: